src/util/color.ts

```diff
@@ -30,59 +32,88 @@ class Color {
     static red: Color;
 
     /**
-     * Parses valid CSS color strings and returns a `Color` instance.
-     * @param input A valid CSS color string.
+     * Parses CSS color strings and converts colors to sRGB color space if needed.
+     * Officially supported color formats:
+     * - keyword, e.g. 'aquamarine' or 'steelblue'
+     * - hex (with 3, 4, 6 or 8 digits), e.g. '#f0f' or '#e9bebea9'
+     * - rgb and rgba, e.g. 'rgb(0,240,120)' or 'rgba(0%,94%,47%,0.1)' or 'rgb(0 240 120 / .3)'
+     * - hsl and hsla, e.g. 'hsl(0,0%,83%)' or 'hsla(0,0%,83%,.5)' or 'hsl(0 0% 83% / 20%)'
+     *
+     * @param input CSS color string to parse.
      * @returns A `Color` instance, or `undefined` if the input is not a valid color string.
      */
-    static parse(input?: string | Color | null): Color | void {
-        if (!input) {
-            return undefined;
+    static parse(input: String | string | undefined | null): Color | undefined {
+        if (Object.prototype.toString.call(input) !== '[object String]') {
```

Comment:
Why call to string? seems a bit expensive, why not use the original `input instanceof Color?

Reply:
The intention behind this function is to parse strings, there should be no situation where an already parsed color is passed to `Color.parse` method, checking if `instanceof Color` in my opinion is beyond the scope of this function. I believe that this is a dead code from ancient times and I took the liberty of removing it.

Support for `String` objects has been added because of the `validateColor` flow, which now uses this function to check if a string is a valid color. Actually this can be avoided by casting `String` into `string` primitive in `validateColor`. So never mind, I'll change it.